.github/workflows/terraform.yaml

```diff
@@ -0,0 +1,156 @@
+        run: |
+          cat<<EOF > step-1/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-1"
+              }
+            }
+          }
+          EOF
+          cat<<EOF > step-2/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-2"
+              }
+            }
+          }
+          EOF
+
+      - name: Deploy
+        id: deploy
+        continue-on-error: true
+        run: ./vast-cloud deploy --auto-approve
+
+      - name: Retry deploy
+        if: steps.deploy.outcome=='failure'
+        run: |
+          echo "Deploy sometimes fails for an unexplained reason."
+          echo "Retrying after a few minutes usually works."
+          sleep 200
+          ./vast-cloud deploy --auto-approve
+
+      - name: Start and restart VAST server
+        run: |
+          echo "Run start-vast-server"
+          ./vast-cloud start-vast-server
+
+          echo "Run start-vast-server again"
+          ./vast-cloud start-vast-server 2> /dev/null \
+            && { echo "Starting server again should fail"; false; } \
+            || true
+          [[ $(./vast-cloud get-vast-server | wc -w) = "1" ]] \
+            || { echo "Only one task should be started"; false; }
+
+          echo "Run restart-vast-server"
+          ./vast-cloud restart-vast-server
+          [[ $(./vast-cloud get-vast-server | wc -w) = "1" ]] \
+            || { echo "Only one task should be started"; false; }
+
+          echo "The task needs a bit of time to boot, sleeping for a while..."
+          sleep 100
+
+      - name: Test db empty from Lambda
+        run: |
+          result=$(./vast-cloud run-lambda -c "vast count")
+          echo "Expected vast count 0, got $result"
+          [[ $result = "0" ]]
+
+      - name: Import data
+        run: |
+          DATA_URL=https://raw.githubusercontent.com/tenzir/vast/master/vast/integration/data/suricata/eve.json
+          ./vast-cloud execute-command \
+            -c "wget -O - -o /dev/null $DATA_URL | vast import suricata"
+
+      - name: Test db not empty from Lambda
+        run: |
+          result=$(./vast-cloud run-lambda -c "vast count")
+          echo "Expected vast count 7, got $result"
+          [[ $result = "7" ]]
+
+      - name: Destroy
+        continue-on-error: true
+        id: destroy
+        if: always()
+        run: ./vast-cloud destroy --auto-approve
+
+      - name: Retry destroy
+        if: steps.destroy.outcome=='failure'
+        run: |
+          echo "Destroy sometimes fails for an unexplained reason."
+          echo "Retrying after a few minutes usually works."
+          sleep 300
+          ./vast-cloud destroy --auto-approve
```

Comment:
This can be simplified a fair bit by simply running in a loop in the run block with a configured amount of retries and sleep intervals. What are the consequences if the retries fail?

Reply:
If the `destroy` doesn't finish successfully, there can be resources remaining that 
1. might cost money
2. might block a deploy, e.g. on the next iteration

Reply:
@dominiklohmann Do you mean a bash loop or is there a specific syntax with Github Action? With the current implementation I do like the fact that you can see from the execution summary what failed or not without looking at the logs. But I do agree that the current formulation is pretty verbose for a simple retry.

Not sure that having more than 1 retry would bring a lot of value, Terraform already implements a pretty comprehensive retry strategy, and it's quite rare that having an extra layer of retries is useful. Here it is necessary because we are using a resource with a specific configuration (Lambda in a VPC) that sometimes behaves unexpectedly and Terraform doesn't manage it well.

Reply:
I think this becomes obsolete if we can fix https://github.com/tenzir/vast/pull/2171#discussion_r841217213, so I don't mind either way until then.